### PR TITLE
AB#4097 Add user service api mock

### PR DESCRIPTION
**Pull request by ESDC-Olivier-Roy**
### Description
Add a user-service api mock to be used on the front end application locally when not using the API locally

### Related Azure Boards Work Items
[AB#4097](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/4097)
### Checklist

- [ ] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/mocks/user-api.server.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }),
+  ),
+});
+
+export function getUserApiMockHandlers() {
+  log.info('Initializing User API mock handlers');
+
+  return [
+    http.post(`${CDCP_API_BASE_URI}/api/v1/users`, async ({ params, request }) => {
+      log.debug('Handling request for [%s]', request.url);
```
Comment (gregory-j-baker) [inline frontend/app/mocks/user-api.server.ts:29]:
```suggestion
      log.debug('Handling POST request for [%s]', request.url);
```

File: frontend/app/mocks/user-api.server.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      if (!parsedUserToCreate.success) {
+        throw new HttpResponse(null, { status: 400 });
+      }
+
+      const userAttributesList = parsedUserToCreate.data.userAttributes.map((userAttribute) => db.userAttributes.create(userAttribute));
+
+      db.user.create({
+        email: parsedUserToCreate.data.email,
+        userAttributes: userAttributesList,
+      });
```
Comment (gregory-j-baker) [inline frontend/app/mocks/user-api.server.ts:47]:
```suggestion
      db.user.create({
        email: parsedUserToCreate.data.email,
        emailVerified: false,
        userAttributes: userAttributesList,
      });
```

File: frontend/app/mocks/user-api.server.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      value: z.string(),
+    }),
+  ),
+});
+
+export function getUserApiMockHandlers() {
+  log.info('Initializing User API mock handlers');
+
+  return [
+    http.post(`${CDCP_API_BASE_URI}/api/v1/users`, async ({ params, request }) => {
```
Comment (nicholas-ly) [inline frontend/app/mocks/user-api.server.ts:28]: If `CDCP_API_BASE_URI` is configured to be the actual API, requests will still hit this mock.

```suggestion
    http.post('https://api.cdcp.example.com/api/v1/users', async ({ params, request }) => {
```

We should also add the ability to enable/disable this API mock. See:
- https://github.com/DTS-STN/canadian-dental-care-plan/blob/main/frontend/app/mocks/node.ts
- https://github.com/DTS-STN/canadian-dental-care-plan/blob/0f4bb6dde6505c9c66ed00164525c21a6c78540d/frontend/.env.example#L203

File: gitops/overlays/training/configs/frontend/config.conf [hunk 4]
```diff
@@ -16,7 +16,7 @@ ENABLED_FEATURES=doc-upload,email-alerts,hcaptcha,view-personal-info,view-applic
 #
 # Global mock configuration
 #
-ENABLED_MOCKS=cct,lookup,power-platform,raoidc,status-check,wsaddress,subscription,application-history
+ENABLED_MOCKS=cct,lookup,power-platform,raoidc,status-check,wsaddress,subscription,application-history,user-api
```
Comment (nicholas-ly) [inline gitops/overlays/training/configs/frontend/config.conf:19]: The changes to the GitOps project should be done in a separate PR.

**Review by nicholas-ly — APPROVED** [review]
